### PR TITLE
The Rust version of the selection sort.

### PR DESCRIPTION
**Pull request by WSL0809**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Comment by krahets** [comment]
To be reviewed by @sjinzh

**Comment by WSL0809** [comment]
and The Rust version of the binary_search_edge

File: codes/rust/chapter_searching/binary_search_edge.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            j = m - 1; // 首个小于 target 的元素在区间 [i, m-1] 中
+        }
+    }
+    if i == nums.len() as i32 || nums[i as usize] != target {
+        return -1; // 未找到目标元素，返回 -1
+    }
+    i
+}
+
+fn binary_search_right_edge(nums: &[i32], target: i32) -> i32 {
```
Comment (krahets) [inline codes/rust/chapter_searching/binary_search_edge.rs:26]: same as above

File: codes/rust/chapter_sorting/selection_sort.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+ * File: selection_sort.rs
+ * Created Time: 2023-05-30
+ * Author: WSL0809 (wslzzy@outlook.com)
+ */
+
+include!("../include/include.rs");
+fn selection_sort(nums: &mut [i32]) {
```
Comment (krahets) [inline codes/rust/chapter_sorting/selection_sort.rs:8]: same

File: codes/rust/chapter_sorting/selection_sort.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            if nums[j] < nums[k] {
+                k = j; // 记录最小元素的索引
+            }
+        }
+        // 将该最小元素与未排序区间的首个元素交换
+        nums.swap(i, k);
+    }
+}
+
+fn main() {
```
Comment (krahets) [inline codes/rust/chapter_sorting/selection_sort.rs:24]: /* Driver Code */

File: codes/rust/chapter_searching/binary_search_edge.rs [hunk 4]
```diff
@@ -0,0 +1,56 @@
+/*
+ * File: binary_search_edge.rs
+ * Created Time: 2023-05-31
+ * Author: WSL0809 (wslzzy@outlook.com)
+ */
+
+fn binary_search_left_edge(nums: &[i32], target: i32) -> i32 {
```
[the remark this answers is not in the document]
Comment (WSL0809) [reply codes/rust/chapter_searching/binary_search_edge.rs:8]: I have updated the code.

File: codes/rust/chapter_sorting/selection_sort.rs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ * Author: WSL0809 (wslzzy@outlook.com)
+ */
+
+include!("../include/include.rs");
+
+/* 选择排序 */
+fn selection_sort(nums: &mut [i32]) {
+    let n = nums.len();
+    // 外循环：未排序区间为 [i, n-1]
+    for i in 0..n - 1 {
```
Comment (sjinzh) [inline codes/rust/chapter_sorting/selection_sort.rs:13]: for i in 0..n-1 {

File: codes/rust/chapter_sorting/selection_sort.rs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        // 内循环：找到未排序区间内的最小元素
+        let mut k = i;
+        for j in i + 1..n {
```
Comment (sjinzh) [inline codes/rust/chapter_sorting/selection_sort.rs:16]: for j in i+1..n {

File: codes/rust/chapter_searching/binary_search_edge.rs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        }
+    }
+    if j < 0 || nums[j as usize] != target {
+        return -1; // 未找到目标元素，返回 -1
+    }
+    j
+}
+
+pub fn main() {
+    /* Driver Code */
```
Comment (sjinzh) [inline codes/rust/chapter_searching/binary_search_edge.rs:48]: /* Driver Code */
pub fn main() {

**Review by sjinzh — CHANGES_REQUESTED** [review]
Request changes.

File: codes/rust/chapter_searching/binary_search_edge.rs (hunk 7 above)
Comment (WSL0809) [reply codes/rust/chapter_searching/binary_search_edge.rs]: Code has been updated.

**Review by krahets — APPROVED** [review]
Thanks for the PR!

**Pull request merged**